### PR TITLE
Fix srcset attribute on images downloaded

### LINKED ISSUE #2981: Grabbed Pages using responsive images display no image in wallabag article
### Issue details

When responsive images are used in the website, wallabag fetches the images, but leaves the srcset paramaters in, which resolves to no image being displayed.

E.g. this site http://www.heise.de/-3603596 resolves by graby to the https://www.heise.de/tp/features/Indische-Realitaeten-3603596.html?view=print and has i.e. the following code

```
<figure class="rte__inlinebild col-lg-12 col-md-12 col-sm-12 col-xs-12" style="max-width:3648px;">
            
            <img src="//1.f.ix.de/scale/geometry/700/q75/tp/imgs/89/2/1/2/7/9/2/3/783b4c1df50cbe0b.jpg" srcset="/scale/geometry/350/q75/tp/imgs/89/2/1/2/7/9/2/3/783b4c1df50cbe0b.jpg 350w, /scale/geometry/700/q75/tp/imgs/89/2/1/2/7/9/2/3/783b4c1df50cbe0b.jpg 700w, /scale/geometry/1050/q75/tp/imgs/89/2/1/2/7/9/2/3/783b4c1df50cbe0b.jpg 1050w, /scale/geometry/2300/q75/tp/imgs/89/2/1/2/7/9/2/3/783b4c1df50cbe0b.jpg 2300w" sizes="(min-width: 80em) 43.75em, (min-width: 64em) 66.66vw, 100vw" alt="" class="img-responsive">
        <figcaption class="rte__inlinebild_source"><span><div class="text">In den Großstädten gibt es für ein mickriges Gehalt Lärmterror und Abgase … (Foto: Gilbert Kolonko)</div></span>
        </figcaption>
</figure>
```

This code resolves after grabbing the content to

```
<figure class="rte__inlinebild col-lg-12 col-md-12 col-sm-12 col-xs-12 c1">
<img src="https://wallabag.asdf.de/assets/images/2/3/238e88b1/9b0f5075.jpeg" srcset="/scale/geometry/350/q75/tp/imgs/89/2/1/2/7/9/2/3/07c122c20e5d8592.jpg 350w, /scale/geometry/700/q75/tp/imgs/89/2/1/2/7/9/2/3/07c122c20e5d8592.jpg 700w, /scale/geometry/1050/q75/tp/imgs/89/2/1/2/7/9/2/3/07c122c20e5d8592.jpg 1050w, /scale/geometry/2300/q75/tp/imgs/89/2/1/2/7/9/2/3/07c122c20e5d8592.jpg 2300w" sizes="(min-width: 80em) 43.75em, (min-width: 64em) 66.66vw, 100vw" alt="" class="img-responsive">
<figcaption class="rte__inlinebild_source"><div class="text">… und das eine oder andere Feuerchen zu löschen. Foto: Gilbert Kolonko</div>
```

If you now open the article in wallabag, no image is displayed, although it is even cached on the server

### Environment

* wallabag version (or git revision) that exhibits the issue: 2.2.2
* How did you install wallabag? Via `git clone` or by downloading the package? git
* Last wallabag version that did not exhibit the issue (if applicable):
* php version: 5
* OS: Debian
* type of hosting (shared or dedicated): dedicated
* which storage system you choose at install (SQLite, MySQL/MariaDB or PostgreSQL): mysql

### Steps to reproduce/test case

grab the above mentioned page and have a look at it in wallabag, you see no picture

### PR DESCRIPTION
**Pull request by Simounet**
Hi there,
A long time issue bothering me. This is a fix for the `srcset` image attribute handling. Should fix #2981 and #3217. Every image variation handled. Keep in mind that the storage size might be a lot bigger.
Have a nice day.

### CODE REVIEW

File: tests/Wallabag/CoreBundle/Helper/DownloadImagesTest.php [hunk 1]
```diff
@@ -183,4 +183,25 @@ public function testProcessRealImage()
...
+
+        $client->getEmitter()->attach($mock);
+
+        $logHandler = new TestHandler();
+        $logger = new Logger('test', [$logHandler]);
+
+        $download = new DownloadImages($client, sys_get_temp_dir() . '/wallabag_test', 'http://wallabag.io/', $logger);
+        $res = $download->processHtml(123, '<p><img class="alignnone wp-image-1153" src="http://piketty.blog.lemonde.fr/files/2017/10/F1FR-530x375.jpg" alt="" width="628" height="444" srcset="http://piketty.blog.lemonde.fr/files/2017/10/F1FR-530x375.jpg 530w, http://piketty.blog.lemonde.fr/files/2017/10/F1FR-768x543.jpg 768w, http://piketty.blog.lemonde.fr/files/2017/10/F1FR-900x636.jpg 900w" sizes="(max-width: 628px) 100vw, 628px" /></p>', 'http://piketty.blog.lemonde.fr/2017/10/12/budget-2018-la-jeunesse-sacrifiee/');
+
+        $this->assertNotContains('http://piketty.blog.lemonde.fr/', $res, 'Image srcset attribute were not replaced');
```
[the remark this answers is not in the document]
Comment (Simounet) [reply tests/Wallabag/CoreBundle/Helper/DownloadImagesTest.php]: Yep, the test name was wrong. I fixed it.

**Comment by Simounet** [comment]
Sorry about that. It must be ok now.

File: tests/Wallabag/CoreBundle/Helper/DownloadImagesTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (j0k3r) [reply tests/Wallabag/CoreBundle/Helper/DownloadImagesTest.php]: I was talking about the _Image srcset attribute were not replaced_. Shouldn't this be _Image srcset attribute were replaced_ instead?

[the remark this answers is not in the document]
Comment (Simounet) [reply tests/Wallabag/CoreBundle/Helper/DownloadImagesTest.php]: Sorry being a little foggy today… It shouldn't contain `http://piketty.blog.lemonde.fr/`. If found, it wasn't replaced correctly.

[the remark this answers is not in the document]
Comment (j0k3r) [reply tests/Wallabag/CoreBundle/Helper/DownloadImagesTest.php]: Forgot about what I said, you're right

**Comment by j0k3r** [comment]
Looks like you have a CS: https://travis-ci.org/wallabag/wallabag/jobs/386519276#L1023

**Comment by Simounet** [comment]
I can do it! 💪

**Comment by j0k3r** [comment]
👍

**Comment by nhoizey** [comment]
👍🏻

**Pull request merged**